x/profiles/legacy/v300/store.go

```diff
@@ -0,0 +1,112 @@
+package v300
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	v230 "github.com/desmos-labs/desmos/v2/x/profiles/legacy/v230"
+	"github.com/desmos-labs/desmos/v2/x/profiles/types"
+)
+
+// MigrateStore performs in-place store migrations from v2.3 to v3.0.
+// The migration includes:
+//
+// - replace all relationship subspace id from string to uint64
+// - replace all user blocks subspace id from string to uint64
+func MigrateStore(ctx sdk.Context, storeKey sdk.StoreKey, cdc codec.BinaryCodec) error {
+	store := ctx.KVStore(storeKey)
+
+	err := migrateUserBlocks(store, cdc)
+	if err != nil {
+		return err
+	}
+
+	err = migrateRelationships(store, cdc)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// migrateUserBlocks migrates the user blocks stored to the new type, converting the subspace from string to uint64
+func migrateUserBlocks(store sdk.KVStore, cdc codec.BinaryCodec) error {
```

Comment:
This is done due to the fact that, looking the `Iterator` interface documentation we can see the following: 
> No writes can happen to a domain while there exists an iterator over it, some backends may take
> out database locks to ensure this will not happen.

So we first need to iterate over the various keys, get the values and then once the iterator is closed we can update the domain. I just noticed I need to close it before the second for loop, will fix it